module/os/linux/zfs/abd_os.c

```diff
@@ -631,17 +632,17 @@ void
 abd_update_scatter_stats(abd_t *abd, abd_stats_op_t op)
 {
 	ASSERT(op == ABDSTAT_INCR || op == ABDSTAT_DECR);
+	int waste = P2ROUNDUP(abd->abd_size, PAGESIZE) - abd->abd_size;
 	if (op == ABDSTAT_INCR) {
 		ABDSTAT_BUMP(abdstat_scatter_cnt);
 		ABDSTAT_INCR(abdstat_scatter_data_size, abd->abd_size);
-		ABDSTAT_INCR(abdstat_scatter_chunk_waste,
-		    P2ROUNDUP(abd->abd_size, PAGESIZE) - abd->abd_size);
+		ABDSTAT_INCR(abdstat_scatter_chunk_waste, waste);
+		arc_space_consume(waste, ARC_SPACE_ABD_CHUNK_WASTE);
```

Comment:
This will account for waste for all consumers of abd and not just the ARC so won't we end up over inflating arc_size?

Reply:
It does account for all consumers of abd's.

Reply:
My initial concern was that new abd consumers could negatively impact the `arc_size` so we may want to document that `arc_size` is not just what is stored in the ARC but any waste that can occur from any ABD consumer.

Reply:
Yeah that's true.  I checked the other consumers of `abd_alloc()`, and they should all be pretty short-lived, so it shouldn't add up to much.  I'll add a comment to `arc_space_consume()`, unless you had somewhere else in mind.